dali/pipeline/operators/expressions/arithmetic_meta.h

```diff
@@ -336,6 +347,25 @@ inline ArithmeticOp NameToOp(const std::string &op_name) {
   return it->second;
 }
 
+inline DALIDataType TypeNameToTypeId(const std::string &type_name) {
```

Comment:
The names you used _almost_ coincide with names used in DALI pipeline. Either make them less similar (i8, i16, i32, u8, u16, ..., f16, f32, f64) or, if you want them to match, perhaps move the functionality of gettint DALIDataType by name to TypeTable so the names are always in sync.

Reply:
Is it big of a difference between u16/i16 and uint16/int16? I am not sure if we want to tie that with DALIDataType. I can move it to expression parsing, as it is specified in the grammar for the expressions.